crates/optimism/payload/src/builder.rs

```diff
@@ -634,6 +698,22 @@ where
                 continue
             }
 
+            // TODO: if interop is active, validate the transaction with the supervisor
+
+            if self.chain_spec.is_interop_active_at_timestamp(self.attributes().timestamp()) {
+                let (is_valid, _is_recoverable) = is_cross_tx_valid::<Evm::Primitives>(
+                    tx.inner(),
+                    self.supervisor.as_ref().unwrap(),
+                    self.supervisor_safety_level.unwrap(),
+                    self.attributes().timestamp(),
+                );
```

Comment:
most of these params are accessible via self, so make this a method instead of stand alone func